src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -248,10 +248,43 @@ private static void BindProperties(object instance, IConfiguration configuration
 
             foreach (PropertyInfo property in modelProperties)
             {
-                BindProperty(property, instance, configuration, options);
+                if (constructorParameters is null || !constructorParameters.Any(p => p.Name == property.Name))
+                {
+                    BindProperty(property, instance, configuration, options);
+                }
+                else
+                {
+                    ResetPropertyValue(property, instance, options);
```

Comment:
Is this just here for compatibility to make sure we still call the setters for types that have constructor parameters?

Reply:
That is right. Look at https://github.com/dotnet/runtime/blob/06ac04f5a946a4db281405d616f89d8f44bb2d6b/src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.TestClasses.cs#L170 as example that can break if we don't do that.